### PR TITLE
Adding documentation for status and error codes

### PR DESCRIPTION
**Pull request by jbrendel**

### CODE REVIEW

File: STATUS-ERROR-CODES.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Summary of status- and error-codes to be returned by an implementation
+
+The follow HTTP status codes should be used by the ITSM REST implementation.
```
Comment (Wade-GitHub) [inline STATUS-ERROR-CODES.md:3]: `following`

File: STATUS-ERROR-CODES.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+Successful POST requests (note that the new resource's URL needs to be
+contained in the "Location" header in the response).
+
+## Client errors
+
+Note that any '4**' error code indicates that it's possible for the client
+to fix the situation by changing something in their request. Thus, for any
+'4**' error, the return body should contain a human readable error message
+explaining the problem. The maintainers of the client code can see those
+messages and can use it to improve the cient implementation.
```
Comment (Wade-GitHub) [inline STATUS-ERROR-CODES.md:23]: `client`

File: STATUS-ERROR-CODES.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+In general, any requests that received a '4**' response will NOT be automatically
+retried by the client, but may be retried at a later time if and when manual fixes
+have been applied to the client code or configuration.
+
+### 400: Bad Request
+
+If there is something wrong with the request body (in case of POST or PUT),
+or if the query parameters (for some GET requests) are wrong. This includes
+situation where the request is correctly formatted (correct syntax), but
```
Comment (Wade-GitHub) [inline STATUS-ERROR-CODES.md:33]: `situations`

File: STATUS-ERROR-CODES.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+where the request is still wrong in a sematic sense. An example here is the
```
Comment (Wade-GitHub) [inline STATUS-ERROR-CODES.md:34]: `semantic`

File: STATUS-ERROR-CODES.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+creation of a user/customer association, where the specified association
+exists already.
+
+### 401: Unauthorized
+
+The client could not be authenticated (note that the official name of this
+return code is "Unauthorized", but by convention, it is used to indicate
+that the authentication failed. This can happen if the client tried to access
```
Comment (Wade-GitHub) [inline STATUS-ERROR-CODES.md:42]: I think after the word `failed` is where you should put a closing bracket `)`

File: STATUS-ERROR-CODES.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+implemented carefully by the server:
+
+### 500: Internal Server Error
+
+This could mean any kind of error, permanent or temporary. Since the client doesn't
+know more, it may as well retry.
+    
+### 502: Bad Gateway
+
+This often is caused either by configuration issues and may be retried.
```
Comment (Wade-GitHub) [inline STATUS-ERROR-CODES.md:96]: Remove `either`

Comment (jbrendel) [reply STATUS-ERROR-CODES.md]: done

File: STATUS-ERROR-CODES.md (hunk 5 above)
Comment (jbrendel) [reply STATUS-ERROR-CODES.md]: done

File: STATUS-ERROR-CODES.md (hunk 4 above)
Comment (jbrendel) [reply STATUS-ERROR-CODES.md]: done

File: STATUS-ERROR-CODES.md (hunk 3 above)
Comment (jbrendel) [reply STATUS-ERROR-CODES.md]: done

File: STATUS-ERROR-CODES.md (hunk 2 above)
Comment (jbrendel) [reply STATUS-ERROR-CODES.md]: done

File: STATUS-ERROR-CODES.md (hunk 1 above)
Comment (jbrendel) [reply STATUS-ERROR-CODES.md]: done

**Pull request merged**